crates/matrix-sdk/src/config/client.rs

```diff
@@ -89,11 +94,89 @@ impl Debug for ClientConfig {
     }
 }
 
+#[cfg(feature = "sled_state_store")]
+mod store_helpers {
+    use matrix_sdk_sled::StateStore;
+
+    use super::Result;
+
+    /// Build the sled Store with the default settings - as a temporary storage
+    pub async fn default_store() -> Result<Box<StateStore>> {
+        Ok(Box::new(StateStore::open()?))
+    }
+
+    /// Build a sled store at `name` (being a relative or full path), and open
+    /// the store with the given passphrase (if given) for encryption
+    pub async fn default_store_with_name(
+        name: &str,
+        passphrase: Option<&str>,
+    ) -> Result<Box<StateStore>> {
+        Ok(Box::new(match passphrase {
+            Some(pass) => StateStore::open_with_passphrase(name, pass)?,
+            _ => StateStore::open_with_path(&name)?,
+        }))
+    }
+}
+
+#[cfg(feature = "indexeddb_stores")]
+mod store_helpers {
+    use matrix_sdk_indexeddb::StateStore;
+
+    use super::Result;
+
+    /// Open the IndexedDB store with the default name, unencrypted
+    pub async fn default_store() -> Result<Box<StateStore>> {
+        Ok(Box::new(StateStore::open().await?))
+    }
+
+    /// Open the indexeddb store at `name` (IndexedDB Database name), and open
+    /// the store with the given passphrase (if given) for encryption
+    pub async fn default_store_with_name(
+        name: &str,
+        passphrase: Option<&str>,
+    ) -> Result<Box<StateStore>> {
+        Ok(Box::new(match passphrase {
+            Some(pass) => StateStore::open_with_passphrase(name.to_owned(), pass).await?,
+            _ => StateStore::open_with_name(name.to_string()).await?,
+        }))
+    }
+}
+
+#[cfg(not(any(feature = "indexeddb_stores", feature = "sled_state_store")))]
+mod store_helpers {
+    use matrix_sdk_base::store::MemoryStore as StateStore;
+
+    use super::Result;
+    /// Open a new in-memory StateStore
+    pub async fn default_store() -> Result<Box<StateStore>> {
+        Ok(Box::new(StateStore::new()))
+    }
+
+    /// Alias for `default_store` - in Memory Stores are never named
+    pub async fn default_store_with_name(
+        _name: &str,
+        _passphrase: Option<&str>,
+    ) -> Result<Box<StateStore>> {
+        Ok(Box::new(StateStore::new()))
+    }
+}
+
+pub use store_helpers::{default_store, default_store_with_name};
+
 impl ClientConfig {
     /// Create a new default `ClientConfig`.
-    #[must_use]
-    pub fn new() -> Self {
-        Default::default()
+    pub async fn new() -> Result<Self> {
+        let mut d = Self::default();
+        d.base_config = d.base_config.state_store(default_store().await?);
+        Ok(d)
+    }
+
+    /// Create a new ClientConfig with a named state store, encrypted with the
+    /// given passphrase (if any)
+    pub async fn with_named_store(name: &str, passphrase: Option<&str>) -> Result<Self> {
```

Comment:
Didn't we want to get rid of such a method, because we had one mentioning a path, which doesn't exist in the indexeddb world.

But we instead switched to the `name` concept, won't this confuse the majority of users since they will likely use sled and a path? I wonder if there's a better middle ground between `path` and `name`

Reply:
I only left this to be somewhat compatible to what we had before and many tests went for it and assume that if certain features are activated, certain stores are there by magic. IMHO I'd remove the `ClientConfig` altogether and make it the users job bring stores and http configured - but that would break the "feature enables default"-sort-of-behaviour we have now. If that is cool with you, I'm happy to do it.

Reply:
I think I prefer to have a simpler setup for most people, so let's leave it.